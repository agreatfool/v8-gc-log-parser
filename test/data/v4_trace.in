[28987:0x101804a00]    12759 ms: Scavenge 63.0 (90.9) -> 50.2 (92.9) MB, 1.4 / 0 ms (+ 0.1 ms in 29 steps since last GC) [allocation failure].
[28987:0x101804a00]    12860 ms: Scavenge 64.4 (92.9) -> 51.6 (95.9) MB, 2.7 / 0 ms (+ 0.1 ms in 29 steps since last GC) [allocation failure].
[28987:0x101804a00]    12910 ms: Mark-sweep 58.8 (95.9) -> 51.0 (96.9) MB, 3.8 / 0 ms (+ 8.2 ms in 101 steps since start of marking, biggest step 0.9 ms) [GC interrupt] [GC in old space requested].
