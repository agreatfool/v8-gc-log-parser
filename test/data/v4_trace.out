[
  {
    "pid": 28987,
    "isolate": "0x101804a00",
    "time_since_init": 12759,
    "data": {
      "collector": "scavenge",
      "start_object_size": 63,
      "end_object_size": 50.2,
      "start_memory_size": 90.9,
      "end_memory_size": 92.9,
      "pause": 1.4,
      "external_time": 0,
      "step_description": "+ 0.1 ms in 29 steps since last GC",
      "gc_reason": "allocation failure"
    },
    "type": "trace"
  },
  {
    "pid": 28987,
    "isolate": "0x101804a00",
    "time_since_init": 12860,
    "data": {
      "collector": "scavenge",
      "start_object_size": 64.4,
      "end_object_size": 51.6,
      "start_memory_size": 92.9,
      "end_memory_size": 95.9,
      "pause": 2.7,
      "external_time": 0,
      "step_description": "+ 0.1 ms in 29 steps since last GC",
      "gc_reason": "allocation failure"
    },
    "type": "trace"
  },
  {
    "pid": 28987,
    "isolate": "0x101804a00",
    "time_since_init": 12910,
    "data": {
      "collector": "mark_sweep",
      "start_object_size": 58.8,
      "end_object_size": 51,
      "start_memory_size": 95.9,
      "end_memory_size": 96.9,
      "pause": 3.8,
      "external_time": 0,
      "step_description": "+ 8.2 ms in 101 steps since start of marking, biggest step 0.9 ms",
      "gc_reason": "GC interrupt",
      "collector_reason": "GC in old space requested"
    },
    "type": "trace"
  }
]
